iothub_client/tests/iothub_client_retry_control_ut/iothub_client_retry_control_ut.c

```diff
@@ -925,9 +925,9 @@ TEST_FUNCTION(Should_Retry_EXPONENTIAL_BACKOFF_success)
     unsigned int option_value = 2;
     int set_option_result = retry_control_set_option(handle, RETRY_CONTROL_OPTION_INITIAL_WAIT_TIME_IN_SECS, &option_value);
 
-    int expected_retry_times[] = { 0, 2, 4, 8, 16 };
+    int expected_retry_times[] = { 0, 2, 4, 8, 16, 30, 30 };
```

Comment:
See above: first retry should not be after 0 seconds (that spins the CPU, likely useless if the network is at fault).

Reply:
I would disagree. 0 seconds was added for a reason. In lab tests we got that by the time the SDK took to detect a disconnection (either by a delay on OS notification or no notification at all), the underlying network available again, but the iothub connection was severed. Retrying at 0 seconds showed to be the most effective on most of wired and wifi connection gaps.